SF iOS/SF iOS/Feed/EventsFeedViewController.m

```diff
@@ -70,24 +75,76 @@ - (void)viewDidLoad {
+    [self.tableView.refreshControl addTarget:self action:@selector(handleRefresh) forControlEvents:UIControlEventValueChanged];
     
     if (self.traitCollection.forceTouchCapability == UIForceTouchCapabilityAvailable) {
         [self registerForPreviewingWithDelegate:self sourceView:self.tableView];
     }
     
+    CGRect searchBarRect = CGRectMake(0, 0, self.view.frame.size.width, kSEARCHBARHEIGHT);
+    self.searchBar = [[UISearchBar alloc] initWithFrame:searchBarRect];
+    self.searchBar.searchBarStyle = UISearchBarStyleMinimal;
+    self.searchBar.placeholder = NSLocalizedString(@"Filter", @"Prompt to search for event names. Here, `Filter` is a joke in English because people filter coffee and this list can be filtered by a term.");
+    self.searchBar.showsCancelButton = true;
+    self.searchBar.delegate = self;
+    
+    // TODO: There’s probably a better way of doing this that doesn’t require a new view
+    CGRect tableSearchViewRect = CGRectMake(0, 0, self.searchBar.frame.size.width, kTABLEHEADERHEIGHT);
+    UIView *tableSearchView = [[UIView alloc] initWithFrame:tableSearchViewRect];
+    tableSearchView.backgroundColor = UIColor.whiteColor;
+    [tableSearchView addSubview:self.searchBar];
+    self.tableView.tableHeaderView = tableSearchView;
+    
+    self.tableView.tableHeaderView.backgroundColor = UIColor.whiteColor;
+    CGPoint contentOffest = self.tableView.contentOffset;
+    contentOffest.y += kSEARCHBARHEIGHT;
+    self.tableView.contentOffset = contentOffest;
+    
+    // TODO: Add an invisible, dismissng button below the search UI
+    
+    self.noResultsView = [[UIView alloc] init];
+    self.noResultsView.frame = CGRectMake(0, (1.5 * kTABLEHEADERHEIGHT), self.tableView.frame.size.width, (self.view.bounds.size.height - (4 * kTABLEHEADERHEIGHT)));
+    [self.noResultsView setHidden:true];
+    [self.view addSubview:self.noResultsView];
+    CGRect labelFrame = CGRectMake(0, 0, self.view.frame.size.width, 100);
+    UILabel *noResultsLabel = [[UILabel alloc] initWithFrame:labelFrame];
+    noResultsLabel.text = NSLocalizedString(@"No events", @"Displayed when no Event names match the given search term");
+    noResultsLabel.textAlignment = NSTextAlignmentCenter;
+    noResultsLabel.font = [UIFont preferredFontForTextStyle:UIFontTextStyleTitle2];
+    [self.noResultsView addSubview:noResultsLabel];
+    
+    [noResultsLabel.topAnchor constraintEqualToAnchor:self.noResultsView.topAnchor].active = true;
+    [noResultsLabel.leftAnchor constraintEqualToAnchor:self.noResultsView.leftAnchor].active = true;
+    [noResultsLabel.rightAnchor constraintEqualToAnchor:self.noResultsView.rightAnchor].active = true;
+    [noResultsLabel.bottomAnchor constraintEqualToAnchor:self.noResultsView.bottomAnchor].active = true;
+    [noResultsLabel.widthAnchor constraintEqualToAnchor:self.noResultsView.widthAnchor].active = true;
+
+    
```

Comment:
We don't have a linter in the project but moving forward double spaces may not be advisable. I'll go through and change it if I ever touch this section of code.

Reply:
Done